src/concrete_solve.jl

```diff
@@ -283,7 +291,11 @@ function DiffEqBase._concrete_solve_adjoint(prob, alg,
                     if typeof(_save_idxs) <: Number
                         _out[_save_idxs] = @view(x[_save_idxs])
                     elseif _save_idxs isa Colon
-                        vec(_out) .= vec(x)
+                        if ArrayInterfaceCore.ismutable(u)
```

Comment:
You want to revert the changes to this anonymous function, and just if else whether to do this one or the one below.